receiver/prometheus/options.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import "time"
+
+// Option interface defines for configuration settings to be applied to receivers.
+//
+// WithReceiver applies the configuration to the given receiver.
+type Option interface {
+	WithReceiver(*Receiver)
+}
+
+type metricBufferPeriod struct {
+	period time.Duration
+}
+
+var _ Option = (*metricBufferPeriod)(nil)
+
+func (mfd *metricBufferPeriod) WithReceiver(ocr *Receiver) {
```

Comment:
This is immaterial but I'm curious why the preference for this more verbose functional option pattern instead of shorter (warning "comment compiled"):

```golang
type Option func (*Receiver)

func WithMetricBufferPeriod(period time.Duration) Option {
	return func (r *Receiver) Option {
		r.metricBufferPeriod = period
	}
}

func WithMetricBufferCount(count int) Option {
	return func (r *Receiver) Option {
		r.metricBufferCount = count
} 
```

Reply:
Nice suggestion, this looks cleaner to me. (I copied this `Option` from other receivers.) /cc @odeke-em any specific reason we made `Option` an interface instead of a func?

Reply:
It is an interface to hide whatever method is implementing it but also to avoid clashes with other packages. With the new suggestion above, in any package one can make an Option just by the definition `type Option func(*Receiver)` which would defeat the purpose of making options that are specified to a package and to hide information. With an interface they can never be equal, unless am mistaken.

Reply:
I see. Since the goal is encapsulation I would suggest the interface Option to be defined with the lower-case method instead, i.e.: `withExporter` since the goal is to prevent other packages to implement Option. Side note: the functional pattern doesn't need the Option to be public but then golint doesn't like the With* returning a non-public type...